storm-core/src/jvm/org/apache/storm/ui/UIHelpers.java

```diff
@@ -0,0 +1,319 @@
+                                 String keyPassword, String tsPath, String tsPassword, String tsType, Boolean needClientAuth, Boolean wantClientAuth) {
+        if (port > 0) {
+            server.addConnector(mkSslConnector(port, ksPath, ksPassword, ksType, keyPassword,
+                    tsPath, tsPassword, tsType, needClientAuth, wantClientAuth));
+        }
+    }
+
+    public static FilterHolder corsFilterHandle() {
+        FilterHolder filterHolder = new FilterHolder(new CrossOriginFilter());
+        filterHolder.setInitParameter(CrossOriginFilter.ALLOWED_ORIGINS_PARAM, "*");
+        filterHolder.setInitParameter(CrossOriginFilter.ALLOWED_ORIGINS_PARAM, "GET, POST, PUT");
+        filterHolder.setInitParameter(CrossOriginFilter.ALLOWED_ORIGINS_PARAM, "X-Requested-With, X-Requested-By, Access-Control-Allow-Origin, Content-Type, Content-Length, Accept, Origin");
+        filterHolder.setInitParameter(CrossOriginFilter.ACCESS_CONTROL_ALLOW_ORIGIN_HEADER, "*");
+        return filterHolder;
+    }
+
+    public static FilterHolder mkAccessLoggingFilterHandle() {
+        return new FilterHolder(new AccessLoggingFilter());
+    }
+
+    public static void configFilter(Server server, Servlet servlet, List filtersConfs) {
+        if (filtersConfs != null) {
+            ServletHolder servletHolder = new ServletHolder(servlet);
+            ServletContextHandler context = new ServletContextHandler(server, "/");
+            context.addServlet(servletHolder, "/");
+            context.addFilter(corsFilterHandle(), "/*", EnumSet.allOf(DispatcherType.class));
+            for (Object obj : filtersConfs) {
+                Map filterConf = (Map) obj;
+                String filterName = (String) filterConf.get(keyword("filter-name"));
+                String filterClass = (String) filterConf.get(keyword("filter-class"));
+                Map filterParams = (Map) filterConf.get(keyword("filter-params"));
+                if (filterClass != null) {
+                    FilterHolder filterHolder = new FilterHolder();
+                    filterHolder.setClassName(filterClass);
+                    if (filterName != null) {
+                        filterHolder.setName(filterName);
+                    } else {
+                        filterHolder.setName(filterClass);
+                    }
+                    if (filterParams != null) {
+                        filterHolder.setInitParameters(filterParams);
+                    } else {
+                        filterHolder.setInitParameters(new HashMap<String, String>());
+                    }
+                    context.addFilter(filterHolder, "/*", FilterMapping.ALL);
+                }
+            }
+            context.addFilter(mkAccessLoggingFilterHandle(), "/*", EnumSet.allOf(DispatcherType.class));
+            server.setHandler(context);
+        }
+    }
+
+    public static Map ringResponseFromException(Exception ex) {
```

Comment:
We should probably leave this in clojure and inline it in the files that use it, so when they move to new technology it can go away or the appropriate replacement code can be written.